File: backend/StudentManagement.Domain/Entities/Student.cs
using System.ComponentModel.DataAnnotations;

namespace StudentManagement.Domain.Entities;

public class Student
{
    public Guid Id { get; set; }
    
    [Required]
    [StringLength(100)]
    public string Name { get; set; } = string.Empty;
    
    [Required]
    [EmailAddress]
    [StringLength(100)]
    public string Email { get; set; } = string.Empty;
    
    [Required]
    [StringLength(20)]
    public string RA { get; set; } = string.Empty;
    
    [Required]
    [StringLength(14)]
    public string CPF { get; set; } = string.Empty;
    
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
} 

File: frontend/src/types/student.ts
export interface Student {
  id: string
  name: string
  email: string
  cpf: string
  ra: string
  createdAt: string
  updatedAt: string
}

export interface CreateStudentRequest {
  Name: string
  Email: string
  CPF: string
  RA: string
}

export interface UpdateStudentRequest {
  Name?: string
  Email?: string
  CPF?: string
  RA?: string
} 